views/utils.py

```diff
@@ -1,68 +1,49 @@
 import glob
+import io
 import os
-from dataclasses import dataclass
+import re
+import tokenize
+from dataclasses import dataclass, field
 from pathlib import Path
 from typing import TypeAlias
 
-import libcst as cst
-from libcst.metadata import MetadataWrapper, WhitespaceInclusivePositionProvider
 from mypy import api
 
 ROOT_DIR = Path(__file__).parent.parent
 MYPY_CONFIG = ROOT_DIR / "pyproject.toml"
 
 
 ChallengeName: TypeAlias = str
+CODE_SPLITTER = "\n## End of your code ##\n"
```

Comment:
Let's move these constants to be class members, so that they're closer to the code that use them